server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -739,6 +739,14 @@ func (s *MattermostAuthLayer) implicitBoardMembershipsFromRows(rows *sql.Rows) (
 }
 
 func (s *MattermostAuthLayer) GetMemberForBoard(boardID, userID string) (*model.BoardMember, error) {
+	// No synthetic memberships for guests
+	user, err := s.GetUserByID(userID)
+	if err != nil {
+		return nil, err
+	}
+	if user.IsGuest {
+		return nil, nil
```

Comment:
I would like to see us move away from methods that return `nil,nil`.  It is very common for developers to expect if error=nil then the other value is valid.  

Some alternatives would be to return an error if IsGuest, or return three values (*model.BoardMember, bool, error) where the bool signifies if member was found.

Reply:
Thanks @wiggin77, I was suspecting this might be an anti-pattern. I think I'd personally prefer keeping with the tuple return unless there's some side effect I'm not considering. Any preference on a specific error type or just do an `errors.New`?

Reply:
This is not correct, a Guest can be member of a board, in this case you are saying that doesn't matter what board are you querying that if the member is a guest, it has no membership.

It should check for the guest only in the case of the explicit membership is not found.

Reply:
About what to return, I would expect a "NotFound" error here.

Reply:
Thanks @jespino, good catch.